addons/website/tests/test_performance.py

```diff
@@ -35,13 +35,13 @@ class TestStandardPerformance(UtilPerf):
     def test_10_perf_sql_img_controller(self):
         self.authenticate('demo', 'demo')
         url = '/web/image/res.users/2/image_256'
-        self.assertEqual(self._get_url_hot_query(url), 6)
+        self.assertEqual(self._get_url_hot_query(url), 4)
```

Comment:
Yes, less queries is more good!
So it looks like my comment above (using `website.default_lang_id` instead of `website._get_cached(...)`) has an answer: it actually is cheaper.

Reply:
`_get_cached` was website's way to implement their own cache, maybe it was useful before but using a recent ORM the record is correctly cached thus `_get_cached` was kinda redundant. I'm not entirely sure about it, hopefully some website gurus will be able to confirm my say.

Reply:
Hello, `_get_cached()` was introduced to improve perf of public controller with https://github.com/odoo/odoo/commit/6d6b505598413b0d766a8394c7131351ade64266

It is still needed, you can have a quick look by trying to access for instance `/empty_controller_test` route (install `test_website` module).
It will do 4 SQL queries (including one on website to get the `company_id` to fill the `allowed_company_ids` in your `_frontend_pre_dispatch` overide) instead of 3 if you call `website._get_cached('company_id')`.
And that controller should then be 0 hot_query, not 1 in the test :-)

Reply:
@rdeodoo thanks for the information!